src/v/cluster/topic_table_probe.cc

```diff
@@ -0,0 +1,52 @@
+// Copyright 2022 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "cluster/topic_table_probe.h"
+
+#include "cluster/topic_table.h"
+#include "config/configuration.h"
+#include "prometheus/prometheus_sanitize.h"
+
+namespace cluster {
+
+topic_table_probe::topic_table_probe(const topic_table& topic_table)
+  : _topic_table(topic_table) {}
+
+void topic_table_probe::handle_topic_creation(
+  create_topic_cmd::key_t topic_namespace) {
+    if (
+      config::shard_local_cfg().disable_public_metrics()
+      || ss::this_shard_id() != 0
+      || !_topics_history.insert(topic_namespace).second) {
```

Comment:
This check guards against double registration when a topic is deleted and re-created.